src/cfn-mcp-server/awslabs/cfn_mcp_server/impl/tools/resource_scanner_tools.py

```diff
@@ -0,0 +1,267 @@
+    region: Optional[str] = None,
+) -> Dict[str, Any]:
+    """Start a resource scan for specific AWS resource types or the entire account.
+
+    Args:
+        resource_types: List of AWS resource types to scan (e.g., ["AWS::S3::Bucket", "AWS::RDS::DBInstance"])
+                       Provide an empty list [] to scan the entire account
+        region: AWS region to use (e.g., "us-east-1", "us-west-2")
+
+    Returns:
+        Dict containing the scan information:
+        {
+            "scan_id": The unique identifier for the started scan
+        }
+    """
+    # Prompt user for input if no resource type is provided
+    if resource_types is None:
+        common_resource_types = [
+            'AWS::S3::Bucket',
+            'AWS::EC2::Instance',
+            'AWS::RDS::DBInstance',
+            'AWS::Lambda::Function',
+            'AWS::IAM::Role',
+        ]
+
+        raise PromptUser(
+            'Please specify resource types to scan. Options:\n\n'
+            '1. Provide specific resource types)\n'
+            '2. Provide an empty list [] to scan the entire account\n\n'
+            'Common resource types include:\n'
+            + '\n'.join([f'- {rt}' for rt in common_resource_types])
+            + f'\n\nExample usage:\n'
+            f'- Single resource type: ["AWS::S3::Bucket"]\n'
+            f'- Multiple types: {common_resource_types}\n'
+            f'- Entire account: []'
+        )
+
+    try:
+        cfn_utils = CloudFormationUtils(region=region or 'us-east-1')
+    except Exception as e:
+        raise handle_aws_api_error(e)
+
+    try:
+        scan_id = cfn_utils.start_resource_scan(resource_types)
+        return {
+            'scan_id': scan_id,
+        }
+    except Exception as e:
+        raise handle_aws_api_error(e)
+
+
+async def list_resources_by_filter_impl(
+    resource_identifier: Optional[str] = None,
+    resource_type_prefix: Optional[str] = None,
+    tag_key: Optional[str] = None,
+    tag_value: Optional[str] = None,
+    limit: int = 100,
+    next_token: Optional[str] = None,
+    region: Optional[str] = None,
+) -> Dict[str, Any]:
+    """Simple implementation for listing AWS resources with filtering.
+
+    Args:
+        resource_identifier: Filter by specific resource identifier (e.g., physical ID)
+        resource_type_prefix: Filter by resource type prefix (e.g., "AWS::S3::")
+        tag_key: Filter resources by tag key
+        tag_value: Filter resources by tag value (requires tag_key)
+        limit: Maximum number of resources to return (1-100)
+        next_token: AWS pagination token from previous response
+        region: AWS region to use
+
+    Returns:
+        Dict containing filtered resources and pagination info
+    """
+    # Basic validation
+    if limit < 1 or limit > 100:
+        raise ClientError('Limit must be between 1 and 100')
+
+    try:
+        cfn_utils = CloudFormationUtils(region=region or 'us-east-1')
+
+        # Get latest completed scan
```

Comment:
function should accept an optional scanId and only do this if one is not provided